File: scripts/theme-replace.js
const fs = require('fs');
const path = require('path');

const srcDir = path.join(__dirname, '..', 'src');

const replacements = {
  'bg-\\[#121212\\]': 'bg-theme-card',
  'bg-\\[#1A1A1A\\]': 'bg-theme-surface',
  'border-\\[#262626\\]': 'border-theme-border',
  'border-\\[#404040\\]': 'border-theme-borderLight',
  'hover:bg-\\[#1A1A1A\\]': 'hover:bg-theme-surface',
  'hover:bg-\\[#262626\\]': 'hover:bg-theme-border',
  'hover:border-\\[#404040\\]': 'hover:border-theme-borderLight',
  'text-\\[#FF0033\\]': 'text-theme-primary',
  'bg-\\[#FF0033\\]': 'bg-theme-primary',
};

function walkAndReplace(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      walkAndReplace(fullPath);
    } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.ts')) {
      let content = fs.readFileSync(fullPath, 'utf8');
      let modified = false;
      
      for (const [key, value] of Object.entries(replacements)) {
        const regex = new RegExp(key, 'g');
        if (regex.test(content)) {
          content = content.replace(regex, value);
          modified = true;
        }
      }
      
      // Special replacements that don't use regex for special chars
      // We want to avoid replacing bg-black/90 with bg-theme-bg/90 if that breaks
      // Actually, bg-black alone is safe
      const bgBlackRegex = /bg-black(?!\/)/g;
      if (bgBlackRegex.test(content)) {
        content = content.replace(bgBlackRegex, 'bg-theme-bg');
        modified = true;
      }

      if (modified) {
        fs.writeFileSync(fullPath, content);
        console.log(`Updated ${fullPath}`);
      }
    }
  }
}

walkAndReplace(srcDir);
console.log('Done replacing theme colors.');

File: scripts/theme_replace.py
import os
import re

src_dir = os.path.join(os.path.dirname(__file__), '..', 'src')

replacements = {
    r'bg-\[\#121212\]': 'bg-theme-card',
    r'bg-\[\#1A1A1A\]': 'bg-theme-surface',
    r'border-\[\#262626\]': 'border-theme-border',
    r'border-\[\#404040\]': 'border-theme-borderLight',
    r'hover:bg-\[\#1A1A1A\]': 'hover:bg-theme-surface',
    r'hover:bg-\[\#262626\]': 'hover:bg-theme-border',
    r'hover:border-\[\#404040\]': 'hover:border-theme-borderLight',
    r'text-\[\#FF0033\]': 'text-theme-primary',
    r'bg-\[\#FF0033\]': 'bg-theme-primary',
}

def walk_and_replace(directory):
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith('.tsx') or file.endswith('.ts'):
                filepath = os.path.join(root, file)
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                original_content = content
                for pattern, replacement in replacements.items():
                    content = re.sub(pattern, replacement, content)
                
                # Special replacement for bg-black to bg-theme-bg
                # Exclude if it's bg-black/50 or something
                content = re.sub(r'bg-black(?!/)', 'bg-theme-bg', content)
                
                # Special replacement for text-white
                content = re.sub(r'text-white', 'text-theme-text', content)
                
                if content != original_content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(content)
                    print(f"Updated {filepath}")

walk_and_replace(src_dir)
print("Done replacing theme colors.")
